src/Module/Admin/Site.php

```diff
@@ -0,0 +1,685 @@
+		}
+
+		foreach ($storage_backends as $name => $class) {
+			$available_storage_backends[$class] = $name;
+		}
+		unset($storage_backends);
+
+		// build storage config form,
+		$storage_form_prefix = preg_replace('|[^a-zA-Z0-9]|' ,'', $current_storage_backend);
+
+		$storage_form = [];
+		if (!is_null($current_storage_backend) && $current_storage_backend != '') {
+			foreach ($current_storage_backend::getOptions() as $name => $info) {
+				$type = $info[0];
+				$info[0] = $storage_form_prefix . '_' . $name;
+				$info['type'] = $type;
+				$info['field'] = 'field_' . $type . '.tpl';
+				$storage_form[$name] = $info;
+			}
+		}
+
+		$t = Renderer::getMarkupTemplate('admin/site.tpl');
+		return Renderer::replaceMacros($t, [
+			'$title'             => L10n::t('Administration'),
+			'$page'              => L10n::t('Site'),
+			'$submit'            => L10n::t('Save Settings'),
+			'$republish'         => L10n::t('Republish users to directory'),
+			'$registration'      => L10n::t('Registration'),
+			'$upload'            => L10n::t('File upload'),
+			'$corporate'         => L10n::t('Policies'),
+			'$advanced'          => L10n::t('Advanced'),
+			'$portable_contacts' => L10n::t('Auto Discovered Contact Directory'),
+			'$performance'       => L10n::t('Performance'),
+			'$worker_title'      => L10n::t('Worker'),
+			'$relay_title'       => L10n::t('Message Relay'),
+			'$relocate'          => L10n::t('Relocate Instance'),
+			'$relocate_warning'  => L10n::t('Warning! Advanced function. Could make this server unreachable.'),
+			'$baseurl'           => System::baseUrl(true),
+
+			// name, label, value, help string, extra data...
+			'$sitename'         => ['sitename', L10n::t('Site name'), Config::get('config', 'sitename'), ''],
+			'$sender_email'     => ['sender_email', L10n::t('Sender Email'), Config::get('config', 'sender_email'), L10n::t('The email address your server shall use to send notification emails from.'), '', '', 'email'],
+			'$banner'           => ['banner', L10n::t('Banner/Logo'), $banner, ''],
+			'$shortcut_icon'    => ['shortcut_icon', L10n::t('Shortcut icon'), Config::get('system', 'shortcut_icon'), L10n::t('Link to an icon that will be used for browsers.')],
+			'$touch_icon'       => ['touch_icon', L10n::t('Touch icon'), Config::get('system', 'touch_icon'), L10n::t('Link to an icon that will be used for tablets and mobiles.')],
+			'$additional_info'  => ['additional_info', L10n::t('Additional Info'), $additional_info, L10n::t('For public servers: you can add additional information here that will be listed at %s/servers.', get_server())],
+			'$language'         => ['language', L10n::t('System language'), Config::get('system', 'language'), '', $lang_choices],
+			'$theme'            => ['theme', L10n::t('System theme'), Config::get('system', 'theme'), L10n::t('Default system theme - may be over-ridden by user profiles - <a href="/admin/themes">Change default theme settings</a>'), $theme_choices],

```

Comment:
Currently when changing the system theme setting, the admin is directly lead to the configuration of the selected system theme. With your changes one is now directed to the selection of which themes should be activated on the node not the expected configuration of the selected theme,

Reply:
It's true, but since frio theme settings are using a modal (for selecting the background image), it was a modal in a modal, which was unwieldy at best and never actually worked. Now site default theme settings can be comfortably changed from the theme admin (which wasn't entirely possible earlier), whichever current site theme is selected.

Reply:
Could the link then directly go to the admin page of the specific theme and not to the list of available themes on the node?

This would mean one less click and I think is expected from the help text indicating that the link leads directly to the configuration of the system theme.

Reply:
It very well could.